specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2/securitydomain.json

```diff
@@ -0,0 +1,244 @@
+                            }
+                        }
+                    },
+                    "default": {
+                        "description": "Key Vault error response describing why the operation failed.",
+                        "schema": {
+                            "$ref": "common.json#/definitions/KeyVaultError"
+                        }
+                    }
+                },
+                "x-ms-examples": {
+                    "Security domain restore operation": {
+                        "$ref": "./examples/securitydomainrestoreoperation-example.json"
+                    }
+                }
+            }
+        },
+        "/securitydomain/upload/pending": {
+            "get": {
+                "tags": [
+                    "HSMSecurityDomain"
+                ],
+                "operationId": "HSMSecurityDomainUploadPending",
+                "description": "Get Security domain restore operation status",
+                "responses": {
+                    "200": {
+                        "description": "Security domain restore operation status",
+                        "headers": {
+                            "Retry-After": {
+                                "description": "The recommended number of seconds to wait before calling the URI specified in Azure-AsyncOperation.",
+                                "type": "integer"
+                            },
+                            "Azure-AsyncOperation": {
+                                "description": "The URI to poll for completion status.",
+                                "type": "string"
+                            }
+                        },
+                        "schema": {
+                            "type": "object",
+                            "properties": {
+                                "status": {
+                                    "type": "string"
+                                },
+                                "status_details": {
+                                    "type": "string"
+                                }
+                            }
+                        }
+                    },
+                    "default": {
+                        "description": "Key Vault error response describing why the operation failed.",
+                        "schema": {
+                            "$ref": "common.json#/definitions/KeyVaultError"
+                        }
+                    }
+                },
+                "x-ms-examples": {
+                    "Find security domain restore operation status": {
+                        "$ref": "./examples/securitydomainoperationstatus-example.json"
+                    }
+                }
+            }
+        }
+    },
+    "definitions": {
+        "CertificateSet": {
+            "properties": {
+                "certificates": {
+                    "type": "array",
+                    "items": {
+                        "type": "string"
```

Comment:
Shouldn't the certificates (essentially RSA public keys) be using JWA rather than just a string?

Reply:
Using certificates here serves the purpose to more easily identify the keys being used. The resulting security domain will encode the thumbprints of the certificates that were used to create the security domain download. This may be helpful to customers to locate the corresponding keys. JWKs do have a KID property, but different clients may set this property differently and so we would lose this identification scheme.

Reply:
There is room in a JWK to have a sha256 thumbprint, but I think it would take code changes in our library to support it. If did that already in the JWE, and we pull out the thumbprint into the response already to make it more accessible.

Using a JWK would mean around 2 days code changes to support that, and I don't see how it is a benefit to add more layers, and more code.